griptape/tools/query/tool.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from attrs import Factory, define, field
+from schema import Literal, Or, Schema
+
+from griptape.artifacts import BaseArtifact, ErrorArtifact, ListArtifact, TextArtifact
+from griptape.config import config
+from griptape.engines.rag import RagEngine
+from griptape.engines.rag.modules import (
+    PromptResponseRagModule,
+)
+from griptape.engines.rag.rag_context import RagContext
+from griptape.engines.rag.stages import ResponseRagStage
+from griptape.mixins.rule_mixin import RuleMixin
+from griptape.tools.base_tool import BaseTool
+from griptape.utils.decorators import activity
+
+
+@define(kw_only=True)
+class QueryTool(BaseTool, RuleMixin):
+    """Tool for performing a query against data."""
+
+    _rag_engine: RagEngine = field(
+        default=Factory(
+            lambda self: RagEngine(
+                response_stage=ResponseRagStage(
+                    response_modules=[
+                        PromptResponseRagModule(prompt_driver=config.drivers.prompt, rulesets=self.rulesets)
+                    ],
+                ),
+            ),
+            takes_self=True,
+        ),
+        alias="_rag_engine",
+    )
+
+    @activity(
+        config={
+            "description": "Can be used to search through textual content.",
```

Comment:
is text actually the only queryable content?

Reply:
Today, yes.